server/src/test/java/org/elasticsearch/http/HttpStatsTests.java

```diff
@@ -27,13 +34,52 @@ public void testMerge() {
         assertEquals(merged.getTotalOpen(), first.getTotalOpen() + second.getTotalOpen());
         assertThat(merged.getClientStats(), hasSize(first.getClientStats().size() + second.getClientStats().size()));
         assertEquals(merged.getClientStats(), Stream.concat(first.getClientStats().stream(), second.getClientStats().stream()).toList());
+        final Map<String, HttpRouteStats> m = new HashMap<>(first.httpRouteStats());
+        second.httpRouteStats().forEach((k, v) -> m.merge(k, v, HttpRouteStats::merge));
+        assertEquals(merged.httpRouteStats(), m);
+    }
+
+    public void testToXContent() {
+        final var requestSizeHistogram = new long[28];
+        requestSizeHistogram[2] = 9;
+        requestSizeHistogram[4] = 10;
+
+        final var responseSizeHistogram = new long[28];
+        responseSizeHistogram[3] = 13;
+        responseSizeHistogram[5] = 14;
+
+        final var responseTimeHistogram = new long[18];
+        responseTimeHistogram[4] = 17;
+        responseTimeHistogram[6] = 18;
+
+        final HttpRouteStats httpRouteStats = new HttpRouteStats(
+            1,
+            ByteSizeUnit.MB.toBytes(2),
+            requestSizeHistogram,
+            3,
+            ByteSizeUnit.MB.toBytes(4),
+            responseSizeHistogram,
+            responseTimeHistogram
+        );
+
+        assertThat(
+            Strings.toString(new HttpStats(1, 2, List.of(), Map.of("http/path", httpRouteStats)), false, true),
+            equalTo(
+                Strings.format(
+                    """
+                        {"http":{"current_open":1,"total_opened":2,"clients":[],"routes":{"http/path":%s}}}""",
+                    Strings.toString(httpRouteStats, false, true)
```

Comment:
I'd like to have some more tests about the serialization of the per-route stats too, particularly checking the boundary cases of the histogram. See e.g. `TransportActionStatsTests`.

Reply:
Yep that's was the plan after I get feedback on the overall approach. It is now added in 860341ec95ca7a4b4410e5ea1ed73e76b0b16627